interfaces/http/routes/v1/assignments/list.go

```
@@ -0,0 +1,94 @@
+	tenantSlug := ctx.Get(constants.TenantHeader)
+	var parsedValidFrom, parsedValinUntil time.Time
+	var genErr error
+	var planId *uuid.UUID
+	var planassignments *pagination.PaginationView[models.PlanAssignment]
+	validFrom := ctx.Query("validFrom")
+	validUntil := ctx.Query("validUntil")
+	planIdOrSlug := ctx.Query("planIdorSlug")
+
+	// Create pagination input
+	paginationInput := pagination.ExtractPaginationFromContext(ctx)
+	c := context.WithValue(ctx.UserContext(), constants.TenantSlugKey, tenantSlug)
```

golangci-lint (1.64.8)

SA1029: should not use built-in type string as key for value; define your own type to avoid collisions

(staticcheck)